[USING] WalletGeneration.md : Correct factual errors, and move some info to new/advanced password chapter (#593)
zkSNACKs/WasabiDoc: docs/using-wasabi/WalletGeneration.md

## Generating the wallet step-by-step

1. Launch Wasabi Wallet.
The very first time you run the software the `Generate Wallet` tab will be open automatically, but you can also access it by clicking on `File -> Generate Wallet` in the menu bar.

2. Name the new wallet precisely to ensure a proper differentiation from wallets created in the future.
This label is not shared with anyone, it is only stored locally on your computer.

3. Write a long and random password and _back it up_.

> It encrypts your secrets, and you will need it every time you want to spend bitcoin from this wallet, or recover your wallet.<br>

Reviewer: ```suggestion
It encrypts your secrets, and you will need it every time you want to spend bitcoin from this wallet, or recover your wallet.

```

![](/TestPassword.png)

## Creating your wallet password

Wasabi integrates [BIP 38: Password-Protected Private Key](/using-wasabi/BIPs.md#bip-38-password-protected-private-key), which means that the secrets needed to spend the bitcoin are encrypted on the computer.
If someone has compromised your operating system and hardware and he only has the encrypted secrets, then no bitcoin can be spent by him.
You need **both** the encrypted secrets, and the password in order to enable the private key which can sign a spending transaction.
This means that the password is your last line of defense against anyone who tries to steal your bitcoin.

:::tip
A good password or passphrase should be both randomly constructed, and long.

> It is the last line of defense between an attacker and your Bitcoins!

Author: Funny...I first had this as "bitcoins, then changed it at the last minute to "Bitcoins" because I thought it was agreed to use capitalization for the word if not in a transaction context.
I'm OK with it either way, but curious what others here would say.

> It is the last line of defense between an attacker and your Bitcoins!

Reviewer 2: bitcoin = 100 mil satoshis, always in the context of quantitative money.
Bitcoin = the network, used in any context of something other than the money.

> It is the last line of defense between an attacker and your Bitcoins!

Author: Ah, that makes more sense than what I was thinking.

### What not to do

Here are a few examples that do it completely wrong.
You should not generate your password like this:

1. Do not use publicly known information like your grandma's maiden name and the birthday of your dog.
`Emma1992` is a really, really, really bad password, because it can easily be guessed and it is very short.
Here is a list of the worst and [most commonly used passwords](https://en.wikipedia.org/wiki/List_of_the_most_common_passwords#SplashData) in recent years.
Password hackers start with lists like this.

2. Do not use the same password that you have used in other places.
Because if one password is leaked, then other ones are compromised.

3. Do try to use a mix of uppercase and lowercase letters, numbers, and special characters (&$%@ etc.) in your password.

4. Do not take a famous quote, or well-known passage from literature and use it for a password.
It will not fool any serious hacker, even if you throw in substitutions such as: "@" instead of "a", or "$" instead of "s".

### Best practices

> Don't "roll your own" crypto.

Reviewer 2: ```suggestion
:::tip
Don't "roll your own" crypto.
:::

```

![](/TestPassword.png)

## Creating your wallet password

Wasabi integrates [BIP 38: Password-Protected Private Key](/using-wasabi/BIPs.md#bip-38-password-protected-private-key), which means that the secrets needed to spend the bitcoin are encrypted on the computer.
If someone has compromised your operating system and hardware and he only has the encrypted secrets, then no bitcoin can be spent by him.
You need **both** the encrypted secrets, and the password in order to enable the private key which can sign a spending transaction.
This means that the password is your last line of defense against anyone who tries to steal your bitcoin.

:::tip
A good password or passphrase should be both randomly constructed, and long.
It is the last line of defense between an attacker and your bitcoins!
:::

### Password basics

1. Randomness is the single-most important requirement for a strong password, because randomness means that the password has no predictable pattern to it.
This makes it impossible to crack without doing an exhaustive, "brute-force" attack.
2. Assuming a password is constructed randomly, its length has the most impact on its strength.

> However, a password that is very long, but not randomly constructed, such as: "thequickbrownfoxjumpedoverthelazydogsback", or a passage from literature, is certain to be on every serious hacker's guess list.

Reviewer 3: ```suggestion
However, a password that is very long, but not randomly constructed, such as: `thequickbrownfoxjumpedoverthelazydogsback`, or a passage from literature, is certain to be on every serious hacker's guess list.
```

For example, there are 7776 words on the EFF wordlist commonly used with dice to create passphrases.
Plugging 7776 into the calulator to get log2(7776), it is shown that each word taken randomly from the list contributes 12.92 bits of entropy to the strength of the passphrase.
So, a 5-word passphrase selected randomly from the EFF list will have 5 x 12.92 = 64 bits of entropy.

A password constructed with lower and upper-case letters, numbers, and special characters will have log2(95), or 6.57 bits of entropy for each character that comprises the password.
From that, we can now calculate that, in order to get the same level of entropy with a password instead of a 5-word passphrase, the password must be at least 10 characters in length.

The 12 seed words that create a Wasabi wallet come from a standard list of 2084 words.
Knowing that, we can calculate that each word contributes log2(2084), or 11 bits of entropy, giving the entire 12 word seed 132 bits of entropy.
At this level of entropy, an NSA-level search of 1 trillion guesses/second would take, on average, 106,398,614,554,997,456,896 years to correctly guess the seed.

> [Return to Password Basics](/using-wasabi/WalletGeneration.md#password-basics)

Reviewer: ```suggestion
```

Author: I threw this link in for people who, out of curiosity jump from the basic section to the advanced section and then decide they dont want to tackle the advanced material....it's an easy way to get back to where they were.
I'm not attached to either having it there or removing it, but I'd like to hear more opinions on it, if possible.

Reviewer 2: This would be the first time we use such. I think it can be a good idea, though probably better to do this in a more streamlined way across chapters.

Write down the word the number corresponds to, in our case it is `salvo`.
This word by itself is not a good password, it would only take about a thousandth of a second to crack.
Repeat the dice rolling process at least five more times.

After six sets of five rolls, we ended up with `52611` `51631` `63432` `43123` `21641` and `14146`.
This corresponds to the password `salvo rhoda walton mudd croft bride`.

This six-word passphrase, even though easy to remember, is just one of 2.21 x 10^23, or 221 sextillion possible six-word combinations taken from the list of 7776 words.
A computer capable of 1 trillion guesses/second would take, on average, over 3500 years to correctly guess this or any six-word combination of words from the list, if they are chosen randomly.

A 6-word passphrase derived as just discussed has 77 [bits of entropy](/using-wasabi/WalletGeneration.md#advanced-password-analysis).
Best practices for being secure against any adversary [currently recommend](https://blog.securityevaluators.com/understanding-password-complexity-5e0d23643a2a) a minimum of near 80 bits of entropy, so this example phrase barely qualifies.

__Once again:__

> Keep a [secure backup](/using-wasabiColdWasabimd#a-list-of-the-more-common-mediums-of-cold-storage-with-some-of-their-weaknesses) of the password, but store it in a separate location from your main 12 recovery words.

Reviewer 3: ```suggestion
Keep a [secure backup](/using-wasabiColdWasabi.md#a-list-of-the-more-common-mediums-of-cold-storage-with-some-of-their-weaknesses) of the password, but store it in a separate location from your main 12 recovery words.
```

### Advanced password analysis

Assuming a password or passphrase is constructed randomly, its level of entropy can be calculated in this way:

- For a password, the total number of possible outcomes given a specific length (L), and number of characters used (C), is:

> C raised to the power of L, or C^L.<br>

Reviewer 3: Why HTML tags here?

Author: Just for ease of reading, I  wanted to start a new line, beginning with: Example: 

To just insert a blank line between line 162 and line 163 does that, but it also results
in line 163 not being included in the "bullet point block", which I didnt want.

Adding a simple HTML line break seemed to be the simplest answer.

> Example: `jwodnrosqf` is lower-case only (C = 26) and 10 characters long (L = 10). So, 26^10 = 141,000,000,000,000 (141 trillion) or: 1.41 x 10^14.<br>

Author: ```suggestion
Example: `jwodnrosqf` is lower-case only (C = 26) and 10 characters long (L = 10). So, 26^10 = 141,000,000,000,000 (141 trillion).
```

> Including upper-case letters in the same password: `jWoDNrosqF` to double (C), results in 52^10 possible passwords = 145,000,000,000,000,000 (145 quadrillion) or : 1.45 x 10^17.<br>

Author: ```suggestion

```

> Including special characters in the same password: `jW0DNro$qF` raises (C) to 95, resulting in 95^10 possibilities = 59,900,000,000,000,000,000 (59.9 quintillion) or : 5.99 x 10^19.

Author: ```suggestion

```

- For a passphrase chosen randomly from a list of words, the total number of possible outcomes given a specific number of words (N) and wordlist size (S) is: S raised to the power of N, or S^N.<br>
Example: The EFF diceware wordlist contains 7776 words.

> Randomly selecting 5 words from the list results in 7776^5 possible passphrases = 28,400,000,000,000,000,000 (28.4 quintillion) or: 2.84 x 10^19.

Author: ```suggestion
Randomly selecting 5 words from the list results in 7776^5 possible passphrases = 28,400,000,000,000,000,000 (28.4 quintillion).
```

#### How Much Entropy Is Needed

Entropy matters because it allows one to compare the required "search-space" of a potential password with the "search-speed" known to be availble to various potential adversaries.
In doing so, it is possible to then know how long it would take an adversary with known resources to attempt every mathematically possible password or passphrase for a given level of entropy.

Edward Snowden revealed in 2013 that the NSA had the ability to guess passwords at a rate of 1 trillion guesses/second.

> At that speed of guessing, the strongest password in the above examples (59.9 quintillion possibilities) could be guessed in a maximum of 1.9 years.

Author: ```suggestion
At that speed of guessing, the password in the above example (28.4 quintillion possibilities) could be guessed in an average of 5 months.
```

#### "Bits" of entropy

Password entropy is often discussed and compared in a context of `bits of entropy`.
This is, of course, because computers can only process 0's and 1's, or bits, so everything related is expressed that way.
To calculate bits of entropy, use a [calculator that does logarithms](https://miniwebtool.com/log-base-2-calculator/?num1=7776) and compute log2(x), where x = the total number of possible outcomes for an event.

For example, there are 7776 words on the EFF wordlist commonly used with dice to create passphrases.
Plugging 7776 into the calulator to get log2(7776), it is shown that each word taken randomly from the list contributes 12.92 bits of entropy to the strength of the passphrase.
So, a 5-word passphrase selected randomly from the EFF list will have 5 x 12.92 = 64 bits of entropy.

A password constructed with lower and upper-case letters, numbers, and special characters will have log2(95), or 6.57 bits of entropy for each character that comprises the password.
From that, we can now calculate that, in order to get the same level of entropy with a password instead of a 5-word passphrase, the password must be at least 10 characters in length.

> The 12 seed words that create a Wasabi wallet come from a standard list of 2084 words.

Author: ```suggestion

```

> Knowing that, we can calculate that each word contributes log2(2084), or 11 bits of entropy, giving the entire 12 word seed 132 bits of entropy.

Author: ```suggestion

```

> At this level of entropy, an NSA-level search of 1 trillion guesses/second would take, on average, 106,398,614,554,997,456,896 years to correctly guess the seed.

Author: ```suggestion

```
